tests/trainer/test_trainer.py

```diff
@@ -110,17 +110,21 @@ class TestTrainerEquivalence():
     reference_folder: pathlib.Path
 
     def assert_models_equal(self, model_1, model_2):
+        assert id(model_1) != id(model_2), "Same model should not be compared."
```

Comment:
`is` is more standard for identity checks...

```suggestion
        assert model_1 is not model_2, "Same model should not be compared."
```